src/backend/distributed/planner/distributed_planner.c

```diff
@@ -779,6 +784,8 @@ CreateDistributedPlan(uint64 planId, Query *originalQuery, Query *query, ParamLi
 	 */
 	if (IsModifyCommand(originalQuery))
 	{
+		RecordSubPlansUsedInPlan(distributedPlan, originalQuery);
```

Comment:
Maybe create a wrapper called `FinalizeDistributedPlan` that just calls this function. This is mainly to make it easier to understand what's going on here and that for instance, the call to `RecordSubPlansUsedInPlan` is not related to `IsModifyCommand`.